site/src/job_queue.rs

```diff
@@ -24,19 +27,118 @@ async fn enqueue_master_commits(ctxt: &Arc<SiteCtxt>) {
                 &master_commit.parent_sha,
                 pr,
                 master_commit.time,
-                BenchmarkRequestStatus::WaitingForParent,
+                BenchmarkRequestStatus::ArtifactsReady,
                 "",
                 "",
             );
             conn.insert_benchmark_request(&benchmark).await;
         }
     }
+    Ok(())
+}
+
+/// This function requires the sorts the priority order of the queue. It assumes
+/// that the only status is `ArtifactsReady`
+fn sort_benchmark_requests<'a>(
+    done: &mut HashSet<String>,
+    unordered_queue: &'a mut [BenchmarkRequest],
+) -> &'a [BenchmarkRequest] {
+    // A topological sort, where each "level" is additionally altered such that
+    // try commits come first, and then sorted by PR # (as a rough heuristic for
+    // earlier requests).
+
+    // Ensure the queue is ordered by status and the `commit_type`
+    unordered_queue.sort_unstable_by_key(|bmr| (bmr.status.rank(), bmr.commit_type.rank()));
+
+    let mut finished = 0;
+    while finished < unordered_queue.len() {
+        // The next level is those elements in the unordered queue which
+        // are ready to be benchmarked (i.e., those with parent in done or no
+        // parent).
+        let level_len = partition_in_place(unordered_queue[finished..].iter_mut(), |bmr| {
+            bmr.parent_sha().is_none_or(|parent| done.contains(parent))
+        });
+
+        // No commit is ready for benchmarking. This can happen e.g. when a try parent commit
+        // was forcefully removed from the master branch of rust-lang/rust. In this case, just
+        // let the commits be benchmarked in the current order that we have, these benchmark runs
+        // just won't have a parent result available.
+        if level_len == 0 {
+            return unordered_queue;
+        }
+
+        let level = &mut unordered_queue[finished..][..level_len];
+        level.sort_unstable_by_key(|bmr| {
+            (
+                bmr.parent_sha().is_some(),
```

Comment:
This is not required, we only assume try/master `ArtifactsReady` here and they must have a parent. We shouldn't ever sort release artifacts here, these will be sorted separately. We should sort instead by the commit type - master before try.